src/translators/crate.py

```diff
@@ -701,6 +701,75 @@ def query(self,
             result.extend(entities)
         return result
 
+    def query_ids(self,
+              entity_type=None,
+              from_date=None,
+              to_date=None,
+              limit=10000,
+              offset=0,
+              fiware_service=None,
+              fiware_servicepath=None):
+        if limit == 0:
+            return []
+
+        where_clause = self._get_where_clause(None,
+                                              from_date,
+                                              to_date,
+                                              fiware_servicepath,
+                                              None)
+
+        if entity_type:
+            table_names = [self._et2tn(entity_type, fiware_service)]
+        else:
+            table_names = self._get_et_table_names(fiware_service)
+
+        if fiware_service is None:
+            for tn in table_names:
+                if "." in tn:
+                    table_names.remove(tmp)
+
+        limit = min(10000, limit)
+        offset = max(0, offset)
+        len_tn = 0
+        result = []
+        stmt = ''
+        for tn in table_names:
+            len_tn += 1
+            if len_tn != len(table_names):
+                stmt += "select entity_id, entity_type, max(time_index) " \
+                           "from {tn} {where_clause}" \
+                           "group by entity_id, entity_type " \
```

Comment:
Like I said in an earlier comment, I don't think grouping by `entity_type` is necessary since all rows should have the same value. This is a potential performance hot spot. If it's an easy thing to do, I'd just replace:

    select entity_id, entity_type, max(time_index)

with

    select entity_id, '<current entity type>' as entity_type, max(time_index)

where `current entity type` comes from the meta table you scan (if it's there, can't remember) or you could just

    (select entity_type from {tn} limit 1) as entity_type

Just an idea. But don't bother if this is a train smash for you. We're going to run a performance optimisation task at some point in time, we can easily tackle this and other QL issues then.

Reply:
Oh forgot to mention. The reason I said it could be a train smash, is that you'd have to verify my theory is actually correct and the query adjustment I suggested is better than what you have right now in terms of query execution plan, response time, memory usage, etc.